File: codewars/6kyu/bit-counting/dduck-bit-counting.py
# bit-counting
# https://www.codewars.com/kata/526571aae218b8ee490006f4

from unittest import TestCase

countBits = lambda n: len(list(filter(lambda x: x == "1", list(bin(n)))))

TestCase().assertEqual(countBits(0), 0);
TestCase().assertEqual(countBits(4), 1);
TestCase().assertEqual(countBits(7), 3);
TestCase().assertEqual(countBits(9), 2);
TestCase().assertEqual(countBits(10), 2);

# best practice in codewars
# def countBits(n):
#     return bin(n).count("1")

File: codewars/6kyu/bit-counting/dduck-bit-counting.js
// bit-counting
// https://www.codewars.com/kata/526571aae218b8ee490006f4

const countBits = (n) => n.toString(2).split('').filter((v) => v === '1').length;
// const countBits = (n) => n !== 0 ? n.toString(2).match(/1/g).length : 0;

console.log(countBits(0), 0);
console.log(countBits(4), 1);
console.log(countBits(7), 3);
console.log(countBits(9), 2);
console.log(countBits(10), 2);
